ch_tools/chadmin/cli/replica_group.py

```diff
@@ -0,0 +1,265 @@
+)
+@option(
+    "-n",
+    "--dry-run",
+    is_flag=True,
+    default=False,
+    help="Enable dry run mode and do not perform any modifying actions.",
+)
+@pass_context
+def restart_replica_command(ctx, on_cluster, dry_run, **kwargs):
+    """
+    Restart one or several table replicas.
+    """
+    cluster = get_cluster_name(ctx) if on_cluster else None
+    for replica in list_table_replicas(ctx, **kwargs):
+        restart_table_replica(
+            ctx,
+            replica["database"],
+            replica["table"],
+            cluster=cluster,
+            dry_run=dry_run,
+        )
+
+
+@replica_group.command("restore")
+@option_group(
+    "Replica selection options",
+    option(
+        "-a",
+        "--all",
+        "_all",
+        is_flag=True,
+        help="Filter in all replicas.",
+    ),
+    option(
+        "-d",
+        "--database",
+        "database_pattern",
+        help="Filter in replicas to restore by the specified database name pattern."
+        " The value can be either a pattern in the LIKE clause format or a comma-separated list of items to match.",
+    ),
+    option(
+        "--exclude-database",
+        "exclude_database_pattern",
+        help="Filter out replicas to restore by the specified database name pattern."
+        " The value can be either a pattern in the LIKE clause format or a comma-separated list of items to match.",
+    ),
+    option(
+        "-t",
+        "--table",
+        "table_pattern",
+        help="Filter in replicas to restore by the specified table name pattern."
+        " The value can be either a pattern in the LIKE clause format or a comma-separated list of items to match.",
+    ),
+    option(
+        "--exclude-table",
+        "exclude_table_pattern",
+        help="Filter out replicas to restore by the specified table name pattern."
+        " The value can be either a pattern in the LIKE clause format or a comma-separated list of items to match.",
+    ),
+)
+@option(
+    "--cluster",
+    "--on-cluster",
+    "on_cluster",
+    is_flag=True,
+    help="Restore replicas on all hosts of the cluster.",
+)
+@option(
+    "-n",
+    "--dry-run",
+    is_flag=True,
+    default=False,
+    help="Enable dry run mode and do not perform any modifying actions.",
+)
+@pass_context
+def restore_command(ctx, _all, on_cluster, dry_run, **kwargs):
+    """
+    Restore one or several table replicas.
+    """
+    cluster = get_cluster_name(ctx) if on_cluster else None
+    ro_replicas = list_table_replicas(ctx, is_readonly=True, **kwargs)
+    for replica in ro_replicas:
+        try:
+            restore_table_replica(
+                ctx,
+                replica["database"],
+                replica["table"],
+                cluster=cluster,
+                dry_run=dry_run,
+            )
+        except ClickhouseError as e:
+            msg = e.response.text
+            if "Replica has metadata in ZooKeeper" in msg or "NO_ZOOKEEPER" in msg:
```

Comment:
It will be good to add a warn log message here for these cases

Reply:
Agree. Added.